indirect_test.cc

```diff
@@ -45,6 +45,12 @@ CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
 
 namespace {
 
+#if XYZ_INDIRECT_IS_TRIVIALLY_RELOCATABLE
```

Comment:
This has the problem that you're using `std::is_trivially_relocatable_v` but haven't checked `__cpp_lib_trivially_relocatable` yet. I.e., this would fail if compiled on an implementation where the _compiler_ supports `[[trivially_relocatable]]` and `__is_trivially_relocatable` but the _library_ doesn't implement `is_trivially_relocatable_v` and `uninitialized_relocate`.
So I suggest that this should instead be guarded by
```
#if defined(__cpp_lib_trivially_relocatable)
```
and just eliminate the `XYZ_INDIRECT_IS_TRIVIALLY_RELOCATABLE` macro entirely.

Reply:
Also, if you have any tests for non-`std::allocator` allocators, it would be good to check here that one of those is _not_ trivially relocatable. For example https://p1144.godbolt.org/z/TTG1r6d71
```
using A = boost::interprocess::allocator<int,
  boost::interprocess::managed_shared_memory::segment_manager>;
static_assert(!std::is_trivially_relocatable_v<xyz::indirect<int, A>>);
```